content/docs/command-reference/config.md

```diff
@@ -238,6 +238,23 @@ to learn more about the state file (database) that is used for optimization.
   When an entry in the database is used (e.g. during the `dvc status`), DVC
   updates the timestamp on that entry. This way, when the database needs a
   cleanup, DVC can sort entries chronologically, and remove the oldest ones.
+  
+- `state.dir` - specify a custom location for the state file. This is
+  necessary when using DVC on NFS or other mounted volumes.
+  (By default, store SQLite-based remote indexes and state's
+  `links` and `md5s` caches in the repository itself to avoid any
+  possible state corruption in 'shared cache dir' scenario.) 
```

Comment:
@meierale can you clarify what this long parenthesis means? Does it intend to document the default behavior when this is not set?

Reply:
If this is about default behavior, I thin kit would be best to document that in https://dvc.org/doc/user-guide/project-structure/internal-files instead.

Reply:
BTW https://dvc.org/doc/user-guide/project-structure/internal-files doesn't mention "state file" so probably a terminology update is needed in this doc or in that other one? I.e. what internal files does `state.dir` affect exactly? (I guess `.dvc/tmp/md5s` and `.dvc/tmp/links`) Thanks

Reply:
> If this is about default behavior, I thin kit would be best to document that in https://dvc.org/doc/user-guide/project-structure/internal-files instead.

Yes that's correct in my opinion: the default behavior of dvc (and its internal sqlite use) is to store the caches of `links` and `md5s` internally inside the .dvc folder of the project. I think it does makes sense indeed to document that on the ìnternal-files` page.
This default behavior however may cause problems if the project (its local folder location) resides on a NFS or some other mounted volume that is not directly attached to the users machine. (e.g. `dvc init` or other commands will fail due to some weird SQLite error.)
Thus the PR and this documentation ;-)

Reply:
> BTW https://dvc.org/doc/user-guide/project-structure/internal-files doesn't mention "state file" so probably a terminology update is needed in this doc or in that other one? I.e. what internal files does `state.dir` affect exactly? (I guess `.dvc/tmp/md5s` and `.dvc/tmp/links`) Thanks

correct 👍

Reply:
> I think it does makes sense indeed to document that on the internal-files page.

Please edit that file when you can and I'll help finalize the text in both places with some Github suggestions 🙂 although if you become unavailable np, just please lmk and we'll take it over when possible.

> > internal-files doesn't mention "state file" so probably a terminology update is needed
>
> correct 👍

OK so for now can remove this whole defaults parenthesis probably, since the section already links to the Internal Files doc where that's explained. Thanks

```suggestion
```